internal/controller/managedcluster_controller_test.go

```diff
@@ -106,7 +115,40 @@ var _ = Describe("ManagedCluster Controller", func() {
 					Spec: hmc.ManagementSpec{},
 				}
 				Expect(k8sClient.Create(ctx, management)).To(Succeed())
+				management.Status = hmc.ManagementStatus{
+					AvailableProviders: hmc.ProvidersTupled{
+						InfrastructureProviders: []hmc.ProviderTuple{
+							{
+								Name: "aws",
+							},
+						},
+					},
+				}
+				Expect(k8sClient.Status().Update(ctx, management)).To(Succeed())
 			}
+			By("creating the custom resource for the Kind Credential")
+			err = k8sClient.Get(ctx, typeNamespacedName, credential)
```

Comment:
getting the deliberately wrong-named object does not make much sense (not only there but everywhere except the `mcl` resource), so maybe it'd be better to create the object ignoring the `AlreadyExist` error (which is not mandatory having removal in the `AfterEach` node)

do we actually expect integration tests for the `Credential` resource? If no **and** the tests work without such a resource's object, maybe it might be better just skip that part or leave a todo comment to implement tests on the actual business logic (like a single-node checking only a small piece of logic in the [`internal/controller/template_controller_test.go:199`](https://github.com/Mirantis/hmc/blob/b7e1f2247edf70dff71d45e0c56433b71d561063/internal/controller/template_controller_test.go#L199-L316))

Reply:
I don't think it's needed now for the credentials controller, since most logic for it actually in the managedcluster.

Reply:
Correct, I meant that if the tests in the scope of the managed-ctrl are not supposed to be implemented, then it'd be great to leave a TODO comment. And the point about getting the wrong-named object: I do not understand why the initially misused code should be copied further instead of being fixed in the new pieces of code.

Either way, it is up to you, I just pointed out that the current code is being misused again and again and may be simplified. But I believe that will be resolved during the tests implementation